README.md

```diff
@@ -20,224 +20,237 @@
 
-def preproc(d: Document):
-    return (d.load_uri_to_image_blob()  # load
-             .set_image_blob_normalization()  # normalize color 
-             .set_image_blob_channel_axis(-1, 0))  # switch color axis
-docs = DocumentArray.from_files('img/*.jpg').apply(preproc)
+from docarray import Document, DocumentArray
+from jina import Executor, requests
 
-import torchvision
-model = torchvision.models.resnet50(pretrained=True)  # load ResNet50
-docs.embed(model, device='cuda')  # embed via GPU to speedup
-
-q = (Document(uri='img/00021.jpg')  # build query image & preprocess
-     .load_uri_to_image_blob()
-     .set_image_blob_normalization()
-     .set_image_blob_channel_axis(-1, 0))
-q.embed(model)  # embed
-q.match(docs)  # find top-20 nearest neighbours, done!
-```
+class ImageEmbeddingExecutor(Executor):
 
-Done! Now print `q.matches` and you'll see the URIs of the most similar images.
+    @requests
+    def embedding(self, docs: DocumentArray, **kwargs):
+        docs.apply(self.preproc) # preprocess images
+        model = torchvision.models.resnet50(pretrained=True)  # load ResNet50
+        docs.embed(model, device='cuda')  # embed via GPU to speed up
+        return docs
 
-<p align="center">
-<a href="https://docs.jina.ai"><img src="https://github.com/jina-ai/jina/blob/master/.github/images/readme-q-match.png?raw=true" alt="Print q.matches to get visual similar images in Jina using ResNet50" width="50%"></a>
-</p>
+    def preproc(self, d: Document):
+        return (d.load_uri_to_image_tensor()  # load
+                .set_image_tensor_shape((200, 200))  # resize all to 200x200
+                .set_image_tensor_normalization()  # normalize color
+                .set_image_tensor_channel_axis(-1, 0))  # switch color axis for the PyTorch model later
+```
 
-Add three lines of code to visualize them:
+Done! Now we can use the `ImageEmbeddingExecutor` to generate embeddings for the dataset:
 
 ```python
-for m in q.matches:
-    m.set_image_blob_channel_axis(0, -1).set_image_blob_inv_normalization()
-q.matches.plot_image_sprites()
+da = DocumentArray.from_files('~/Downloads/left/*.jpg') # load the left images from the dataset, adjust the path as needed
+executor = ImageEmbeddingExecutor()
+da_with_embeddings = executor.embedding(da[:10]) # generate the embeddings for the first ten pictures using the Executor
+print(da_with_embeddings.embeddings)
 ```
 
-<p align="center">
-<a href="https://docs.jina.ai"><img src="https://github.com/jina-ai/jina/blob/master/.github/images/cat-similar.png?raw=true" alt="Visualize visual similar images in Jina using ResNet50" width="50%"></a>
-</p>
+### Use Vector Embeddings in a Flow to find similar images
 
-Sweet! FYI, you can use Keras, ONNX, or PaddlePaddle for the embedding model. Jina supports them well.
+With a few additions we can use the embeddings from the previous step in a `Flow` to find similiar images. This can used locally immediately to serve HTTP request with a REST API:
```

Comment:
```suggestion
With a few additions, we can use the embeddings from the previous step in a `Flow` to find similar images. This can be used locally immediately to serve HTTP request with a REST API:
```